apps/web/components/events/UpcomingEventsView.tsx

```diff
@@ -1,18 +1,38 @@
-import { Event } from "@our-scene/api-hooks/resources/events/types";
-import { EventCard } from "./EventCard";
+import { Event } from '@our-scene/api-hooks/resources/events/types';
+import { EventCardPreview } from './EventCardPreview';
+import { useState, useEffect } from 'react';
+import { EventCardDetailed } from './DetailedEventCard';
 
 interface UpcomingEventProps {
   data: Event[];
 }
 
 export const UpcomingEvents = ({ data }: UpcomingEventProps) => {
+  const [selectedEvent, setSelectedEvent] = useState<Event | null>(null);
+
+  const handleEventSelect = (event: Event) => {
+    setSelectedEvent(event);
+  };
+
+  useEffect(() => {
+    setSelectedEvent(data[0]);
+  }, [data]);
+
   return (
-    <div className="grid grid-cols-3">
-      {data.map((event: Event) => (
-        <div key={event.id}>
-          <EventCard event={event} />
+      <div className="grid h-full grid-cols-2 gap-4 py-10 overflow-y-scroll">
+        <div className="flex flex-col px-2 space-y-2">
+          {selectedEvent &&
+            data.map((event: Event) => (
+              <div key={event.id}>
+                <EventCardPreview
+                  event={event}
+                  handleSelect={handleEventSelect}
+                  isFocused={event.id === selectedEvent.id}
+                />
+              </div>
+            ))}
```

Comment:
this is a little strange, imo.

are you checking for `selectedEvent` to confirm that `data` is truthy here? (this also pertains to the useEffect above)


I think what might be less overhead is only rendering this component if there's data available. Otherwise have a loading state at the view level.

that way you can just 

```js
const [selectedEvent, setSelectedEvent] = useState<Event | null>(data[0]);
```

and get rid of the useEffect and the `selectedValue` truthiness check on line 24.

Reply:
I totally agree that this was a bandaid over a larger problem. The `selectedEvent` truthiness check was my (quick) fix to ensure that these only render if data is available and won't break the entire events page. I've updated the events index file so that the query's `isSuccess` status is truthy before rendering the UpcomingEventsView component.